### PR TITLE
Sample extensions for the work item form

### PR DESCRIPTION
**Pull request by sferg-msft**

### CODE REVIEW

File: ui/workItemGroup.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+            // Get the WorkItemFormService.  This service allows you to get/set fields/links on the 'active' work item (the work item
+            // that currently is displayed in the UI).
+            function getWorkItemFormService()
+            {
+                return _WorkItemServices.WorkItemFormService.getService();
+            }
+    
+            // Register a listener for the work item group contribution.
+            VSS.register("Fabrikam.samples-ui.sample-work-item-form-group", function () {
```
Comment (willsmythe) [inline ui/workItemGroup.html]: Does this actually work? I'd expect you to register with "sample-work-item-form-group" (and not include the publisher + extension ID).

File: ui/workItemGroup.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                return {
+                    // Called when the active work item is modified
+                    onFieldChanged: function(args) {
```
Comment (willsmythe) [inline ui/workItemGroup.html]: Does `onFieldChanged` get called? All the functions here seem to be the same as the functions for a form-observer. Does the handler for a form group have any of its own methods?

File: ui/workItemGroup.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                        $(".events").append($("<div/>").text("onFieldChanged - " + JSON.stringify(args)));
+                    },
+                    
+                    // Called when a new work item is being loaded in the UI
+                    onLoaded: function (args) {
```
Comment (willsmythe) [inline ui/workItemGroup.html]: Is this called when the group is loaded or when the entire work item form is loaded?

File: ui/workItemPage.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+            // Get the WorkItemFormService.  This service allows you to get/set fields/links on the 'active' work item (the work item
+            // that currently is displayed in the UI).
+            function getWorkItemFormService()
+            {
+                return _WorkItemServices.WorkItemFormService.getService();
+            }
+
+            // Register a listener for the work item page contribution
+            VSS.register("Fabrikam.samples-ui.sample-work-item-form-page", function () {
```
Comment (willsmythe) [inline ui/workItemPage.html]: See previous comment -- should the first parameter be `sample-work-item-form-page`?

File: ui/workItemGroup.html (hunk 1 above)
Comment (sferg-msft) [inline ui/workItemGroup.html]: Yep, it works.  I looks weird though, but we're just using the id returned by the contribution service on the server here.  Is there a reason why the contribution service isn't returning the shorter id?

File: ui/workItemGroup.html (hunk 2 above)
Comment (sferg-msft) [inline ui/workItemGroup.html]: A group and page is also an observer.  There are no page/group specific methods.

File: ui/workItemGroup.html (hunk 3 above)
Comment (sferg-msft) [inline ui/workItemGroup.html]: the onLoaded event is called when a work item is loaded.  So if you are switching between work items, you will get the event.  It is not an "extension" onLoaded event.

File: ui/workItemPage.html (hunk 4 above)
Comment (sferg-msft) [inline ui/workItemPage.html]: Same answer as earlier.

**Pull request merged**